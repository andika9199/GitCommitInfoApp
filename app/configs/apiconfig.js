import base64 from 'react-native-base64';
import {AsyncStorage} from 'react-native';
const API_GITHUB = 'https://api.github.com/';
const API_LOGIN_GITHUB = API_GITHUB + 'user';
const API_ALL_USER_REPO = API_LOGIN_GITHUB + '/repos';
const API_OTHER_REPO = API_GITHUB + 'repos/';

export async function getToken() {
  try {
    let userData = await AsyncStorage.getItem('loginData');
    let data = await JSON.parse(userData);
    return data;
  } catch (error) {
    return null;
  }
}

export async function loginGit(payload) {
  const encodedData = payload.isLogedIn
    ? payload.isLogedIn
    : `Basic ${base64.encode(`${payload.username}:${payload.password}`)}`;
  const headers = {
    Authorization: encodedData,
  };
  const options = {
    method: 'GET',
    headers,
  };
  try {
    let response = await fetch(API_LOGIN_GITHUB, options);
    let data = await response.json();
    await AsyncStorage.setItem('loginData', JSON.stringify(encodedData));
    return data;
  } catch (error) {
    return 'error';
  }
}

export async function searchRepo(payload) {
  try {
    let response = await fetch(
      API_OTHER_REPO +
        payload.keyword +
        '/commits' +
        `?per_page=30&page=${payload.page}`,
    );
    let data = await response.json();
    return data;
  } catch (error) {
    return 'error';
  }
}
