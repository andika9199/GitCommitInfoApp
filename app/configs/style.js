export const BLACK = '#333333';
export const GRAY_LIGHT = '#cccccc';
export const GRAY_DARKER = '#808080';
export const FOREST_GREEN = '#228B22';
export const GREEN = '#1e7b1e';
export const WHITE = '#ffffff';

export const ITEMS_CENTER = {
  alignItems: 'center',
  justifyContent: 'center',
};

export const DEFAULT_TEXT = {
  fontSize: 13,
  fontFamily: 'Robot',
  color: BLACK,
};

export const DEFAULT_TEXT_BOLD = {
  fontSize: 13,
  fontFamily: 'Robot',
  color: BLACK,
  fontWeight: 'bold',
};

export const TEXT_SMALL = {
  fontSize: 10,
  fontFamily: 'Robot',
  color: BLACK,
  fontWeight: 'bold',
};

export const TEXT_LARGE_BOLD = {
  fontSize: 16,
  fontFamily: 'Robot',
  color: BLACK,
  fontWeight: 'bold',
};

export const TEXT_LARGE = {
  fontSize: 16,
  fontFamily: 'Robot',
  color: BLACK,
};

export const TEXT_BIG = {
  fontSize: 18,
  fontFamily: 'Robot',
  color: BLACK,
  fontWeight: 'bold',
};
