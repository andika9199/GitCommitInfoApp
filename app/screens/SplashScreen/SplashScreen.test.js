import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import configureMockStore from 'redux-mock-store';
import ComponentConnect, {SplashScreen, mapDispatchToProps} from './index';
import {getToken} from '../../configs/apiconfig';

const mockStore = configureMockStore();

jest.mock('../../configs/apiconfig', () => ({
  getToken: jest.fn(),
}));

const initialProps = {
  login: {
    data: null,
    error: false,
    isFetching: false,
  },
  reqLogin: jest.fn(),
  navigation: {
    replace: jest.fn(),
  },
};
describe('Splash Screen', () => {
  describe('Splash Screen rendered', () => {
    it('Screen Renderer Successfully user not Login', async () => {
      await getToken.mockImplementationOnce(() => {
        return null;
      });
      const component = renderer.create(<SplashScreen {...initialProps} />);
      expect(component).toMatchSnapshot();
    });
    it('Screen Renderer Successfully user Login', async () => {
      await getToken.mockImplementationOnce(() => {
        return 'Basic ynwadwadsue==';
      });
      const component = renderer.create(<SplashScreen {...initialProps} />);
      expect(component).toMatchSnapshot();
    });
  });
  describe('Redux', () => {
    const initialState = {
      login: {
        data: {},
        error: false,
        isFetching: false,
      },
    };
    const store = mockStore(initialState);
    const component = shallow(<ComponentConnect store={store} />);

    it('mapStateToProps login should return true', () => {
      expect(component.props().login).not.toEqual({
        data: {},
        isFetching: false,
        error: false,
      });
    });
    it('mapDispatchToProps reqLogin call right type', () => {
      const dispatch = jest.fn();
      mapDispatchToProps(dispatch).reqLogin();
      expect(dispatch.mock.calls[0][0]).toEqual({
        payload: {
          username: undefined,
          password: undefined,
          isLogedIn: undefined,
        },
        type: 'REQ_LOGIN',
      });
    });
  });
});
