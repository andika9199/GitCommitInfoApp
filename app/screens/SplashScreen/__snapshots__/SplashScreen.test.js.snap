// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Splash Screen Splash Screen rendered Screen Renderer Successfully user Login 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../app/assets/gitIcons.png",
      }
    }
    style={
      Object {
        "height": 270,
        "resizeMode": "contain",
        "width": 270,
      }
    }
  />
  <Text
    style={
      Object {
        "color": "#333333",
        "fontFamily": "Robot",
        "fontSize": 18,
        "fontWeight": "bold",
      }
    }
  >
    Monitor commit of the Repo
  </Text>
</View>
`;

exports[`Splash Screen Splash Screen rendered Screen Renderer Successfully user not Login 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../app/assets/gitIcons.png",
      }
    }
    style={
      Object {
        "height": 270,
        "resizeMode": "contain",
        "width": 270,
      }
    }
  />
  <Text
    style={
      Object {
        "color": "#333333",
        "fontFamily": "Robot",
        "fontSize": 18,
        "fontWeight": "bold",
      }
    }
  >
    Monitor commit of the Repo
  </Text>
</View>
`;
